src/org/spdx/htmltemplates/LicenseHTMLFile.java

```diff
@@ -72,23 +75,23 @@ public String getSite() {
 		}
 
 		public boolean getIsValid() {
-			return UrlHelper.urlValidator(url);
```

Comment:
Rather than adding these into the templates and writers, it would be much more efficient to update the URLs in the builder while filling in all the details from the License XML file.  Then you could replace each of the calls to either the static classes or the CrossRefHelper with calls to getCrossRefs from the license.

There are a few different hooks we could use in the builder:

1. Build the crossRefs as we are building seeAlso array at https://github.com/spdx/LicenseListPublisher/blob/f8cb27d6da741af0a926a7ba401b3450909c10e0/src/org/spdx/licensexml/LicenseXmlDocument.java#L207
2. Build the crossRefs just before returning the ListedLicense at https://github.com/spdx/LicenseListPublisher/blob/f8cb27d6da741af0a926a7ba401b3450909c10e0/src/org/spdx/licensexml/LicenseXmlDocument.java#L250
3. Create a new `ISpdxListedLicenseProvider` which simply wraps an existing `ISpdxListedLicenseProvider` which simply calls the `getSeeAlso` method and generates and sets the crossRefs.

The first approach is the most straightfoward, but least flexible.

The third approach is the most flexible.  Since we've talked about possibly making this an option, I would prefer option 3.  Let me know if you need more context or information for this approach.

Reply:
An example of implementing option 3 above since it was a bit complex to describe:

```
	class generatingLicenseIterator implements Iterator<SpdxListedLicense> {

		@Override
		public boolean hasNext() {
			return innerLicenseIterator.hasNext();
		}

		@Override
		public SpdxListedLicense next() {
			SpdxListedLicense retval = innerLicenseIterator.next();
			retval.setCrossRef(getTheCrossRefs(retval.getSeeAlso()));
			return retval;
		}
		
	}

	/**
	 * @throws SpreadsheetException 
	 * @throws LicenseRestrictionException 
	 * @throws SpdxListedLicenseException 
	 * 
	 */
	public CrossRefGeneratingLicenseProvider(ISpdxListedLicenseProvider innerProvider) throws LicenseRestrictionException, SpreadsheetException, SpdxListedLicenseException {
		innerLicenseIterator = innerProvider.getLicenseIterator();
	}

...
```